### PR TITLE
Allow Select to accept Object Value Type

### PR DESCRIPTION
**Pull request by azizhk**

### CODE REVIEW

File: stories/select.story.tsx [hunk 1]
```diff
@@ -74,56 +82,47 @@ storiesOf("Components/Select", module)
...
+                return true;
+              }
+              return option.label
+                .toLowerCase()
+                .includes(store.state.searchQuery.toLowerCase());
+            })
+            .map(option => (
+              <Option<OptionType>
+                key={option.id}
+                value={option}
```
Comment (azizhk) [inline stories/select.story.tsx]: I guess this is what you wanted @dxtr026

**Comment by ritz078** [comment]
assuming immutability ?

**Comment by ritz078** [comment]
btw calendar shows different because everyday the current day and it's ui changes.

**Comment by azizhk** [comment]
Assuming `options.includes(selected)` gives true when an option is selected.

**Comment by ritz078** [comment]
It means that your object isn't cloned and that's not safe to assume.

File: stories/select.story.tsx (hunk 1 above)
Comment (dxtr026) [reply stories/select.story.tsx]: Yes

**Pull request merged**